examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -35,6 +35,32 @@ BdxOtaSender::BdxOtaSender()
     memset(mFileDesignator, 0, chip::bdx::kMaxFileDesignatorLen);
 }
 
+CHIP_ERROR BdxOtaSender::InitializeTransfer(chip::FabricIndex fabricIndex, chip::NodeId nodeId)
+{
+    if (mInitialized)
+    {
+        // Reset stale connection from the Same Node if exists
+        if ((mFabricIndex.HasValue() && mFabricIndex.Value() == fabricIndex) && (mNodeId.HasValue() && mNodeId.Value() == nodeId))
+        {
+            Reset();
+        }
+        // Prevent a new node connection since another is active
+        else if ((mFabricIndex.HasValue() && mFabricIndex.Value() != fabricIndex) ||
```

Comment:
Do `mFabricIndex` and `mNodeId` need to be optional? We could just initialize/reset to kUndefinedFabricIndex/kUndefinedNodeId and makes the check here a little easier?

Reply:
True! We could use an invalid value. But I guess the whole point of using an "optional" was so we don't have to set aside an invalid value. Luckily in this case the spec already defined invalid values (unless it changes in v2.0 ?).
TLDR; I am fine either way, unless you have a strong preference. :-)

Reply:
I don't feel strongly about it, just thought you could have less checks here for `HasValue()`